pkg/ccl/streamingccl/streamproducer/span_config_event_stream.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package streamproducer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeedcache"
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvsubscriber"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/mon"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/errors"
+)
+
+type spanConfigEventStream struct {
+	execCfg *sql.ExecutorConfig
+	spec    streampb.StreamPartitionSpec
+	mon     *mon.BytesMonitor
+	acc     mon.BoundAccount
+
+	data tree.Datums // Data to send to the consumer
+
+	// Fields below initialized when Start called.
+	rfc         *rangefeedcache.Watcher
+	streamGroup ctxgroup.Group // Context group controlling stream execution.
+	doneChan    chan struct{}  // Channel signaled to close the stream loop.
+	updateCh    chan rangefeedcache.Update
+	errCh       chan error       // Signaled when error occurs in rangefeed.
+	streamCh    chan tree.Datums // Channel signaled to forward datums to consumer.
+	sp          *tracing.Span    // Span representing the lifetime of the eventStream.
```

Comment:
Not going to ask you to do it here necessarily, but now that you've done the "new value generator" implementation" it could be worth spending a few minutes to see if there is a cheap way to unify them.

Namely, looking at the new implementation, I think you could probably have some generic implementation that you had so pass something that implemented `Run(ctx, doneChan, streamCh)`.  Anyway, not essential, but just a thought.

Reply:
Yeah, I'd like to refactor this once I have everything hooked up, including the ingestion side.